src/a4/test/BoardSpaceTest.java

```diff
@@ -0,0 +1,94 @@
+package a4.test;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+
+import a4.domain.*;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class BoardSpaceTest {
+	BoardSpaceFactory test_factory = new BoardSpaceFactory();
```

Comment:
We can do either way. I figured I would utilize the Factory here since we've got the code for it.